### PR TITLE
fix: check performance exist

### PR DESCRIPTION
**Pull request by chalecao**

### CODE REVIEW

**Comment by chalecao** [comment]
fix performance check, if not it will block page render with js error if current browser don't support performance in low device

File: packages/scheduler/src/forks/SchedulerDOM.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ import {enableIsInputPending} from '../SchedulerFeatureFlags';
 
 let getCurrentTime;
 const hasPerformanceNow =
-  typeof performance === 'object' && typeof performance.now === 'function';
+  window.performance && typeof performance === 'object' && typeof performance.now === 'function';
```
Comment (rheh) [inline packages/scheduler/src/forks/SchedulerDOM.js:44]: Question: Where is performance coming from in the latter checks?  If performance is on window should it not be:
```
window.performance && typeof window.performance === 'object' && typeof window.performance.now === 'function';
```

or 

```
const performance = window.performance || undefined;
const hasPerformanceNow = typeof performance === 'object' && typeof performance.now === 'function';
```

File: packages/scheduler/src/forks/SchedulerNoDOM.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ var isHostTimeoutScheduled = false;
 
 let getCurrentTime;
 const hasPerformanceNow =
-  typeof performance === 'object' && typeof performance.now === 'function';
+  window.performance && typeof performance === 'object' && typeof performance.now === 'function';
```
Comment (rheh) [inline packages/scheduler/src/forks/SchedulerNoDOM.js:74]: same question as above here.   Where does performance come from?

Comment (chalecao) [reply packages/scheduler/src/forks/SchedulerNoDOM.js:74]: Yeah, I find the reason. In my case, performance was set 'null' some where, so 
    typeof` performance === 'object'  
check pass, but 
    typeof performance.now === 'function' 
cause an error

File: packages/scheduler/src/forks/SchedulerDOM.js (hunk 1 above)
Comment (chalecao) [reply packages/scheduler/src/forks/SchedulerDOM.js:44]: Yeah, I find the reason. In my case, performance was set null some where, not undefined. it's weird.

**Pull request closed without merging**